include/openthread/platform/settings.h

```diff
@@ -72,6 +72,7 @@ enum
     OT_SETTINGS_KEY_SRP_CLIENT_INFO      = 0x000c, ///< The SRP client info (selected SRP server address).
     OT_SETTINGS_KEY_SRP_SERVER_INFO      = 0x000d, ///< The SRP server info (UDP port).
     OT_SETTINGS_KEY_BR_ULA_PREFIX        = 0x000f, ///< BR ULA prefix.
+    OT_SETTINGS_KEY_BR_ON_LINK_PREFIXES  = 0x0010, ///< BR local on-link prefixes.
```

Comment:
nit:
```suggestion
    OT_SETTINGS_KEY_BR_DEPRECATED_ON_LINK_PREFIXES  = 0x0010, ///< BR deprecated on-link prefixes.
```

Reply:
I did not add "deprecate" here to avoid confusion with deprecated settings key. Also we add our current advertising local on-link to this list (when we start advertising it) so the entries in this list are not all necessarily deprecated prefixes.